### PR TITLE
Smoother release process

### LINKED ISSUE #1401: Release process hiccups
First time I've tried to do a calico-containers release today; hit a few snags as a set of fresh eyes:

- [x] `make release` "what to do next" help doesn't mention mac/windows calicoctl binaries.
- [x] `make release` "what to do next" help doesn't mention the list of sub-component versions to embed in the release note or how to collect that.
- [x] RELEASING.md implies you can release from a non-master branch, but that messes up the automatic version numbers assigned by subsequent releases (if the branch isn't merged back to master before the next release).
- [x] What are the pre-reqs for being able to push to quay and dockerhub?
- [ ] `make clean` deletes the `calico/felix` docker image; this caught me out when I was trying to do an end-to-end build on my system without pushing anything.  Feels wrong to delete an artefact that you didn't `make`.
- [x] If build fails have to delete the tag again before rebuild.
- [x] Final check of version number detected a dirty build and failed the build.  Please can we add a dirty check at the beginning?

### PR DESCRIPTION
**Pull request by gunjan5**
Fixes #1401 

- [x] `make release` "what to do next" help doesn't mention mac/windows calicoctl binaries.
- [x] `make release` "what to do next" help doesn't mention the list of sub-component versions to embed in the release note or how to collect that.
- [x] RELEASING.md implies you can release from a non-master branch, but that messes up the automatic version numbers assigned by subsequent releases (if the branch isn't merged back to master before the next release).
- [x] What are the pre-reqs for being able to push to quay and dockerhub?
- [ ] `make clean` deletes the `calico/felix` docker image; this caught me out when I was trying to do an end-to-end build on my system without pushing anything.  Feels wrong to delete an artefact that you didn't `make`.
- [x] If build fails have to delete the tag again before rebuild.
- [x] Final check of version number detected a dirty build and failed the build.  Please can we add a dirty check at the beginning?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -514,7 +528,17 @@ endif
 	docker run $(NODE_CONTAINER_NAME) calico-felix --version
 	docker run $(NODE_CONTAINER_NAME) libnetwork-plugin -v
 
-	@echo "\nNow push the tag and images. Then create a release on Github and attach the dist/calicoctl binary"
+	@echo "\nNow push the tag and images. Then create a release on Github and attach dist/calicoctl, dist/calicoctl-darwin-amd64, and dist/calicoctl-windows-amd64.exe binaries"
```
Comment (fasaxc) [inline Makefile]: WDYT to wrapping this at about 70 chars so it doesn't split awkwardly on a terminal?

File: Makefile [hunk 2]
```diff
@@ -514,7 +528,17 @@ endif
...
+	@echo "\nAdd release notes for calicoctl and calico/node. Use this URL to find commit messages for this release: https://github.com/projectcalico/calicoctl/compare/<old_release_version>...$(VERSION)"
```
Comment (fasaxc) [inline Makefile]: Could do `git log --oneline <old_release_version>...$(VERSION)` locally or some variant of that?

Comment (gunjan5) [inline Makefile]: that's neat! I didn't know about it

**Comment by gunjan5** [comment]
@fasaxc made the changes you suggested. There's just one thing I need your input on, which is at the last comment in #1401

File: RELEASING.md [hunk 3]
```diff
@@ -15,18 +15,29 @@ Creating a new release creates the following artifacts:
 ## Preparing for a release
 1. Make sure you are on the master branch and don't have any local uncommitted changes. e.g. Update the libcalico-go pin to the latest release in glide.yaml and run `glide up -v`, create PR, ensure test pass and merge.
 
-2. Login using your dockerhub credentials (`docker longin` in your terminal). Make sure you have write access to calico orgs on Dockerhub and quay.io. 
+2. Pre-requisits for pushing container images:
+  - Make sure you have write access to calico orgs on Dockerhub and quay.io. 
+  - Login using your dockerhub credentials.
+  - `docker longin` in your terminal.
```
Comment (fasaxc) [inline RELEASING.md]: Typo: longin

File: RELEASING.md [hunk 4]
```diff
@@ -15,18 +15,29 @@ Creating a new release creates the following artifacts:
...
 
 3. Update the sub-component versions in the `Makefile` 
-- `CONFD_VER := v0.10.0-scale`
-- `BIRD_VER := v0.2.0`
-- `GOBGPD_VER := v0.1.1`
-- `FELIX_VER := 2.0.0`
-- `LIBNETWORK_PLUGIN_VER := v1.0.0`
-- `LIBCALICOGO_VER := v1.0.0`
-- `LIBCALICO_VER := v0.19.0`- Currently, the startup.py script relies on the Python version of libcalico
+  - `CONFD_VER := v0.10.0-scale`
```
Comment (fasaxc) [inline RELEASING.md]: Might be best not to list these here and make the Makefile the "master" for these.  Then we only have one place to update?

**Comment by fasaxc** [comment]
@gunjan5 Looks like you've got a conflict to resolve.

File: RELEASING.md (hunk 4 above)
Comment (gunjan5) [inline RELEASING.md]: these versions are only meant to be examples. But I see your point about it being confusing.

**Comment by gunjan5** [comment]
@fasaxc made the changes your suggested. Not sure if the build failure is related. Seems to be failing at `FAIL: test_ingress_policy_can_block_through_traffic (tests.st.policy.test_felix_gateway.TestFelixOnGateway)`. Any idea why?

File: Makefile [hunk 5]
```diff
@@ -18,19 +18,30 @@ CURL=curl -sSf
 SOURCE_DIR?=$(dir $(lastword $(MAKEFILE_LIST)))
 SOURCE_DIR:=$(abspath $(SOURCE_DIR))
 ###############################################################################
+# Subcomponent versions:
+CONFD_VER := v0.10.0-scale
+BIRD_VER := v0.2.0
+GOBGPD_VER := v0.1.1
+FELIX_VER := 2.0.0
```
Comment (fasaxc) [inline Makefile]: The failing test needs felix version `2.0.1-rc1`, `2.0.0` doesn't contain the relevant feature.

**Comment by fasaxc** [comment]
@gunjan5 It's an incorrect merge, you reverted the felix version to `2.0.0`; master uses `2.0.0-rc1`

**Comment by gunjan5** [comment]
that fixed it @fasaxc . Thanks!

**Comment by fasaxc** [comment]
@gunjan5 looks like you've got new conflicts.  Please ping me on slack once resolved and I'll get this merged before it has further time to rot.

**Comment by gunjan5** [comment]
@fasaxc there are drastic changes in master Makefile structure, so I will open a new PR since Makefiles seems to be split into 3 Makefiles now

**Pull request closed without merging**